cores/esp32/esp32-hal-tinyusb.c

```diff
@@ -661,6 +665,13 @@ esp_err_t tinyusb_enable_interface(tinyusb_interface_t interface, uint16_t descr
         log_e("Interface %s invalid or already enabled", (interface >= USB_INTERFACE_MAX)?"":tinyusb_interface_names[interface]);
         return ESP_FAIL;
     }
+    if(interface == USB_INTERFACE_HID && reserve_endpoints){
+        // Some simple PC BIOS requires specific endpoint addresses for keyboard at boot
+        if(!tinyusb_reserve_out_endpoint(1) ||!tinyusb_reserve_in_endpoint(1)){
```

Comment:
Without this reservation of endpoints IN 1, OUT 1, USBCDC will reserve OUT 3, IN 4, IN 5 (line 665), then when `USBHID.cpp:tusb_hid_load_descriptor()` calls `tinyusb_get_free_in_endpoint()`/`tinyusb_get_free_out_endpoint()`, those functions will return IN 3, OUT 4 because *those functions prefer returning endpoints whose opposite is already used*.

Some alternatives to calling `tinyusb_reserve_out_endpoint()`/`tinyusb_reserve_in_endpoint()` here to reserve IN 1, OUT 1:

1. `USBHID.cpp:tusb_hid_load_descriptor()` could call `tinyusb_get_free_duplex_endpoint()` which would likely get endpoints IN 1, OUT 1 because `USBHID.cpp:tusb_hid_load_descriptor()` runs early enough.
2. Change `tinyusb_get_free_in_endpoint()`/`tinyusb_get_free_out_endpoint()` not to prefer returning endpoints whose opposite is already used. Or add a flag parameter.